examples/with-ant-design/app/antd.tsx

```diff
@@ -0,0 +1,15 @@
+'use client'
+
+import { useState, type PropsWithChildren } from 'react'
+import { createCache, extractStyle, StyleProvider } from '@ant-design/cssinjs'
+import { useServerInsertedHTML } from 'next/navigation'
+
+export const RootStyleRegistry = ({ children }: PropsWithChildren) => {
+  const [cache] = useState(() => createCache())
+
+  useServerInsertedHTML(() => {
+    return <style dangerouslySetInnerHTML={{ __html: extractStyle(cache) }} />
```

Comment:
The `extractStyle` fn from `@ant-design/cssinjs` produces a string like:

```
<style data-token-hash="abc" data-css-hash="def">...</style><style data-token-hash="ghi" data-css-hash="jkl">...</style>
```

Reference: https://github.com/ant-design/cssinjs/blob/master/src/hooks/useStyleRegister.tsx#L467

So, I believe, injecting that string inside a style tag, would end up producing an output like:
```html
<style>
    <style data-token-hash="abc" data-css-hash="def">...</style><style data-token-hash="ghi" data-css-hash="jkl">...</style>
</style>
```

I'm not sure about the impact tho.

Reply:
Atm, I'm using https://www.npmjs.com/package/html-react-parser to transform the string returned from  `extractStyle` into react elements.

Reply:
Thank you very much for finding this. Now I have fixed it. There is no extra package installed, but through a little hack code.

Reply:
Pretty neat. 😅